Mention Kotlin DSL script compilation performance improvements in release notes

eskatos opened the pull request:
As a follow up to https://github.com/gradle/gradle/pull/14276

[Rendered version](https://builds.gradle.org/repository/download/Gradle_Check_BuildDistributions/38060722:id/distributions/gradle-6.8-docs.zip!/gradle-6.8-20200912130338%2B0000/docs/release-notes.html#performance-improvements)

pioterj commented on subprojects/docs/src/docs/release/notes.md:
```diff
@@ -32,6 +32,20 @@ details of 2
+
+<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAvIAAAHTCAYAAABFg/xCAAAqA0lEQVR42u3dC5iV9X0ncGNiktq1Juai2Ta7bTdt2u1u202fZtPd7bNb0zXR6BOjVWOqjVk00YTmYoJcRFBBNCJEgkSQijdugnjXqigKiiAC4nCVm+AMEAYFBobLMMBv/b19zmQY5swMKnAGP5/n+T4yc86cc+ad4//3nfe8552jAgAA6HSOsgkAAECRBwAAFHkAAECRBwAARR4AAFDkAQAARR4AABR5AABAkQcAABR5AABQ5AEAAEUeAABQ5AEAAEUeAAAUeQAAQJEHAAAUeQAAUOQBAABFHgAAUOQBAECRBwAAFHkAAECRBwAARR4 ...
+
+The impact on your build may vary, but we are happy to ship this improvement that will noticeably shorten your feedback loop when editing build logic.
```
We should focus on the readers rather than ourselves - for example say "you will be happy to see this improvement" rather than "we are happy to ship this improvement".

Another proposal:
"While the impact on your build may vary, most builds can expect a noticeable shortening of build editing feedback loop with Kotlin DSL thanks to this improvement"

eskatos commented on subprojects/docs/src/docs/release/notes.md:
```diff
@@ -32,6 +32,20 @@ details of 2
+
+On a build of a hundred projects with different build scripts the cumulative script compilation time goes from [~50 seconds](https://scans.gradle.com/s/3bg67eccnya5i/performance/configuration?showScriptCompilationTimes) down to [~21 seconds](https://scans.gradle.com/s/y7dw5ekes24ag/performance/configuration?showScriptCompilationTimes) with cold caches and cold daemons.
+It also has the nice effect of reducing memory pressure.
+Garbage collection time goes from [2.6 seconds](https://scans.gradle.com/s/3bg67eccnya5i/performance/build#garbage-collection) down to [1.3 seconds](https://scans.gradle.com/s/y7dw5ekes24ag/performance/build#garbage-collection).
```
It's not as minor as it seems. And the graph shows it so I'd leave it alongside the script compilation time in the text.

pioterj replied:
OK, fair enough regarding leaving it alongside the compilation time. The speed improvement is the main highlight though. The build scans show 1.3s less in garbage collection and 2,5% less peak memory usage. That's not insignificant on its own but the speed improvement is obviously a much bigger deal. So I'd still use "Additionally" when referring to reduced memory pressure.

eskatos commented:
Thanks for the review @pioterj, I think I addressed all your comments, please take another look

[Rendered version](https://builds.gradle.org/repository/download/Gradle_Check_BuildDistributions/38100389:id/distributions/gradle-6.8-docs.zip!/gradle-6.8-20200915080059%2B0000/docs/release-notes.html#performance-improvements)

pioterj commented on subprojects/docs/src/docs/release/notes.md:
```diff
@@ -34,17 +34,17 @@ details of 2
+### Kotlin DSL script compilation is faster and consumes less memory
 
-In this release, the compilation of [Gradle Kotlin DSL](userguide/kotlin_dsl.html) scripts (`*.gradle.kts`) has been improved and is now faster and consumes less memory.
+In this release, the compilation of [Gradle Kotlin DSL](userguide/kotlin_dsl.html) scripts (`*.gradle.kts`) is now faster and consumes less memory.
```
"in this release" and "now" is a bit repetitive. I'd replace "now faster" with "much faster".
